src/compose/test.rs

```diff
@@ -1557,4 +1558,31 @@ mod test {
 
         f32::from_le_bytes(view.try_into().unwrap())
     }
+
+    #[test]
+    fn wgsl_dual_source_blending() {
+        let mut composer = Composer::default();
+        composer.capabilities.set(Capabilities::DUAL_SOURCE_BLENDING, true);
+
+        let module = composer
+            .make_naga_module(NagaModuleDescriptor {
+                source: include_str!("tests/dual_source_blending/blending.wgsl"),
+                file_path: "tests/dual_source_blending/blending.wgsl",
+                ..Default::default()
+            })
+            .unwrap();
+
+        let info = composer.create_validator().validate(&module).unwrap();
+        let wgsl = naga::back::wgsl::write_string(
+            &module,
+            &info,
+            naga::back::wgsl::WriterFlags::EXPLICIT_TYPES,
+        )
+        .unwrap();
+
+        let mut f = std::fs::File::create("wgsl_call_entrypoint.txt").unwrap();
```

Comment:
This printing stuff should be commented out once it's actually output to the file once and it looks sane.